src/Analyser/TypeSpecifier.php

```diff
@@ -519,19 +519,28 @@ public function specifyTypesInCondition(
 			$methodReflection = $scope->getMethodReflection($methodCalledOnType, $expr->name->name);
 			if ($methodReflection !== null) {
 				$referencedClasses = $methodCalledOnType->getObjectClassNames();
```

Comment:
I think this can be simplified using `getObjectClassReflections` instead of `getObjectClassNames`

(same for static call)